aws-samples/aws-modern-application-workshop
# Dougsch/go mod4

*Issue #, if available:*

*Description of changes:*

Port of Python module 4. Adds support for:

GET
/misfits                                               returns all misfits
/misfits?filter=FILTER&value=VALUE    returns a misfit where FILTER is has VALUE
/misfits/{mysfitsId}                               returns a misfit by their MysfitId

POST:
/misfits/<mysfitId>/like        increments the likes for misfit with mysfitId
/misfits/<mysfitId>/adopt    enables adopt for misfit with mysfitId

You can test GET using a browse:

http://localhost:8088/misfits/da5303ae-5aba-495c-b5d6-eb5c4a66b941

To test POST, use curl:

curl -X POST 'http://localhost:8088/misfits/da5303ae-5aba-495c-b5d6-eb5c4a66b941/like'

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
